src/python_testing/TC_PWRTL_2_1.py

```diff
@@ -46,30 +46,45 @@ async def test_TC_PWRTL_2_1(self):
 
         endpoint = self.user_params.get("endpoint", 1)
 
+        powertop_attr_list = Clusters.Objects.PowerTopology.Attributes.AttributeList
+        powertop_cluster = Clusters.Objects.PowerTopology
+        attribute_list = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=powertop_cluster, attribute=powertop_attr_list)
+        avail_endpoints_attr_id = Clusters.Objects.PowerTopology.Attributes.ActiveEndpoints.attribute_id
+        act_endpoints_attr_id = Clusters.Objects.PowerTopology.Attributes.AvailableEndpoints.attribute_id
+
         self.print_step(1, "Commissioning, already done")
 
-        if not self.check_pics("PWRTL.S.A0000"):
-            logging.info("Test skipped because PICS PWRTL.S.A0000 is not set")
+        self.print_step(2, "Read AvailableAttributes attribute")
+        if avail_endpoints_attr_id in attribute_list:
+            available_endpoints = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=Clusters.Objects.PowerTopology, attribute=attributes.AvailableEndpoints)
+
+            if available_endpoints == []:
+                logging.info("AvailableEndpoints is an empty list")
+            else:
+                logging.info("AvailableEndpoints: %s" % (available_endpoints))
+                asserts.assert_less_equal(len(available_endpoints), 20,
+                                          "AvailableEndpoints length %d must be less than 21!" % len(available_endpoints))
+
+        else:
+            logging.info('Skipping test as available endpoints attribute ID not in attribute list on DUT')
```

Comment:
```suggestion
            self.mark_current_step_skipped()
```
The more you do of these, the more I'm thinking it's worthwhile to pop these into a guard function, similar to pics_guard, but for features and attributes etc. There are now "has_feature" and "has_attribute" functions, it seems like this might be neater. Not a blocker here, but maybe give it some thought about whether that would make this stuff easier to write for other test authors going forward. We could force the tests to pull a full wildcard at the start of the tests (or at least a wildcard of the global attributes). We're kind of going there with the pics reduction stuff anyway.

Reply:
https://github.com/project-chip/matter-test-scripts/issues/370 - started an issue to discuss this further. We can continue discussing on there and you can resolve this comment once it's done.

Reply:
Hi Cecille,
Thank you, I have used the suggestions you have provided in this test, the Opstate tests, and the ICDM 2 1 test regarding creating the attributes_guard function in the matter_testing_support helper module, and have used the has_attributes function for helping with the attribute checks in the OpstateCommon module in PR [#34290](https://github.com/project-chip/connectedhomeip/pull/34290), please let me know if this will work!
Thank you!